Tests/SuperLinq.Test/ToListTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Test;
+
+public class ToListTests
+{
+	[Fact]
+	public void ToListWithSmallerAmountOfElementsThanLength()
+	{
+		const int Length = 2;
+		var enumerable = Enumerable.Range(0, 1);
+
+		var result = enumerable.ToList(Length);
+
+		Assert.Equal(enumerable, result);
+		Assert.Equal(Length, result.Capacity);
+	}
+
+	[Theory]
+	[InlineData(3)]
+	[InlineData(5)]
+	public void ToListWithHigherAmountOfElementsThanLength(int dataLength)
+	{
+		const int Length = 2;
+		var enumerable = Enumerable.Range(0, dataLength);
+
+		var result = enumerable.ToList(Length);
+
+		Assert.Equal(enumerable, result);
+		Assert.True(result.Capacity > Length);
```

Comment:
This assert is not necessary, as if the capacity is less than length, then the `Assert.Equal()` will fail due to invalid size.

Reply:
This test shows that we don't trim the List. I was also thinking about adding an optional bool (`shouldBeTrimmed = false`) to make it possible to trim the List to the expected length.

Reply:
Trimming is not a behavior used by the corefx `.ToList()`. If the consumer wants to trim after the fact, they can do so, but there is no benefit for us to do so (automatically or based on parameter).